server/src/test/java/org/opensearch/common/util/concurrent/QueueResizableOpenSearchThreadPoolExecutorTests.java

```diff
@@ -23,14 +26,19 @@
  * based on the time taken for each event.
  */
 public class QueueResizableOpenSearchThreadPoolExecutorTests extends OpenSearchTestCase {
-    public void testResizeQueueSameSize() throws Exception {
-        ThreadContext context = new ThreadContext(Settings.EMPTY);
-        ResizableBlockingQueue<Runnable> queue = new ResizableBlockingQueue<>(ConcurrentCollections.<Runnable>newBlockingQueue(), 2000);
+    private QueueResizableOpenSearchThreadPoolExecutor executor;
+    private ResizableBlockingQueue<Runnable> queue;
+    private int measureWindow;
 
+    private void createExecutor(int queueSize) {
         int threads = randomIntBetween(1, 10);
-        int measureWindow = randomIntBetween(100, 200);
-        logger.info("--> auto-queue with a measurement window of {} tasks", measureWindow);
-        QueueResizableOpenSearchThreadPoolExecutor executor = new QueueResizableOpenSearchThreadPoolExecutor(
+        measureWindow = randomIntBetween(100, 200);
+        ThreadContext context = new ThreadContext(Settings.EMPTY);
+        this.queue = new ResizableBlockingQueue<>(
+            ConcurrentCollections.newBlockingQueue(),
+            Objects.requireNonNull(queueSize, "All tests must set a queue size")
```

Comment:
Objects.requireNonNull() is designed for reference types only. For primitive data types, this check would always pass. Did you intend to do something else? may be value check?

Reply:
Good catch! This was leftover from a previous version of the code where it actually made sense.